test/MarketMaker.js

```diff
@@ -196,55 +195,65 @@ contract('marketMaker', function(accounts) {
-    help.debug("checking scenario", JSON.stringify(data));
+    //help.debug("checking scenario", JSON.stringify(data));
 
-    assert.equal(data.month, await mm.getCurrentPeriodIndex());
+    assert.equal(data.marketMakerMonth, await mm.getCurrentPeriodIndex());
     data.marketMakerEthBalance.should.be.bignumber.equal(web3.eth.getBalance(mm.address));
     data.marketMakerLifBalance.should.be.bignumber.equal(await token.balanceOf(mm.address));
 
     new BigNumber(web3.toWei(tokenTotalSupply, 'ether')).
-      minus(data.burnedTokens).
+      minus(data.marketMakerBurnedTokens).
       should.be.bignumber.equal(await token.totalSupply.call());
-    data.burnedTokens.should.be.bignumber.equal(await mm.totalBurnedTokens.call());
+    data.marketMakerBurnedTokens.should.be.bignumber.equal(await mm.totalBurnedTokens.call());
 
-    if (data.month < periods) {
+    if (data.marketMakerMonth < periods) {
       data.marketMakerBuyPrice.should.be.bignumber.equal(await mm.getBuyPrice());
       assert.equal(data.claimablePercentage, parseInt(await mm.getAccumulatedDistributionPercentage()));
     }
 
-    assert.equal(data.month >= periods, await mm.isFinished());
+    assert.equal(data.marketMakerMonth >= periods, await mm.isFinished());
 
-    data.customerEthBalance.should.be.bignumber.equal(web3.eth.getBalance(customer));
-    data.customerLifBalance.should.be.bignumber.equal(await token.balanceOf(customer));
+    data.ethBalances[customerAddressIndex].should.be.bignumber.equal(web3.eth.getBalance(customer));
+    data.balances[customerAddressIndex].should.be.bignumber.equal(await token.balanceOf(customer));
 
-    data.maxClaimableEth.should.be.bignumber.equal(await mm.getMaxClaimableWeiAmount());
+    data.marketMakerMaxClaimableEth.should.be.bignumber.equal(await mm.getMaxClaimableWeiAmount());
 
-    data.totalClaimedEth.should.be.bignumber.equal(await mm.totalWeiClaimed.call());
+    data.marketMakerClaimedEth.should.be.bignumber.equal(await mm.totalWeiClaimed.call());
   };
 
-  it("should go through scenario with some claims and sells on the Market Maker", async function() {
+  it.only("should go through scenario with some claims and sells on the Market Maker", async function() {
     // Create MM with balance of 200 ETH and 100 tokens in circulation,
     const priceFactor = 100000;
 
     token = await simulateCrowdsale(tokenTotalSupply, [tokenTotalSupply], accounts);
 
-    let customer = accounts[1];
+    let customer = accounts[customerAddressIndex];
     let startingMMBalance = new BigNumber(web3.toWei(200, 'ether'));
     const initialBuyPrice = startingMMBalance.dividedBy(help.lif2LifWei(tokenTotalSupply));
 
     let state = {
-      month: 0,
-      burnedTokens: new BigNumber(0),
+      marketMakerMonth: 0,
+      marketMakerPeriods: periods,
+      token: token,
+      initialTokenSupply: help.lif2LifWei(tokenTotalSupply),
+      marketMakerBurnedTokens: new BigNumber(0), // burned tokens in MM, via sendTokens txs
```

Comment:
do we need this? why is important to keep track of the burned tokens on the MM?

Reply:
in the prop-based test we want to keep track of tokens burned in MM and tokens burned in general separately, to assert on both different numbers. We don't really need to keep track of both in this test, but we are migrating to using the commands from the prop-based test which have to update everything for that test, so here we are just adding both to be compatible with those commands

Reply:
cant we use the same value value for both and check that the amount of tokens burned match before and after we burned them outside and inside MM? We dont have a variable on the MM contract that keeps track of the tokens burned, thats why I think we dont need it and it add more complexity to what we want to tests that is just token supply. In fact, we can use only totalSupply that is the only variable we are using to keep track of the total tokens amount.